packages/starlight/user-components/LinkButton.astro

```diff
@@ -0,0 +1,76 @@
+---
+import type { HTMLAttributes } from 'astro/types';
+import { Icons } from '../components/Icons';
+import Icon from './Icon.astro';
+
+interface Props extends Omit<HTMLAttributes<'a'>, 'href'> {
+	href: string | URL;
+	icon?: keyof typeof Icons | undefined;
+	iconPlacement?: 'start' | 'end' | undefined;
+	variant?: 'primary' | 'secondary' | 'minimal';
+}
+
+const {
+	class: className,
+	icon,
+	iconPlacement = 'end',
+	variant = 'primary',
+	...attrs
+} = Astro.props;
+---
+
+<a class:list={['sl-link-button not-content', variant, className]} {...attrs}>
+	{icon && iconPlacement === 'start' && <Icon name={icon} size="1.5rem" />}
+	<slot />
+	{icon && iconPlacement === 'end' && <Icon name={icon} size="1.5rem" />}
+</a>
+
+<style>
+	.sl-link-button {
+		align-items: center;
+		border: 1px solid transparent;
+		border-radius: 999rem;
+		display: inline-flex;
+		font-size: var(--sl-text-sm);
+		gap: 0.5em;
+		line-height: 1.1875;
+		outline-offset: 0.25rem;
+		padding: 0.4375rem 1.125rem;
+		text-decoration: none;
+	}
+
+	.sl-link-button.primary {
+		background: var(--sl-color-text-accent);
+		border-color: var(--sl-color-text-accent);
+		color: var(--sl-color-black);
+	}
+	.sl-link-button.primary:hover {
+		color: var(--sl-color-black);
+	}
+	.sl-link-button.secondary {
+		border-color: inherit;
+		color: var(--sl-color-white);
+	}
+	.sl-link-button.minimal {
+		color: var(--sl-color-white);
+		padding-inline: 0;
+	}
+
+	.sl-link-button :global(svg) {
+		flex-shrink: 0;
+	}
+
+	@media (min-width: 50rem) {
+		.sl-link-button {
+			font-size: var(--sl-text-base);
+			padding: 0.9375rem 1.25rem;
+		}
+	}
+
+	:global(.sl-markdown-content) .sl-link-button {
+		margin-inline-end: 0.5em;
```

Comment:
Your demo page was super helpful!

A bit subjective, but what do you think of increasing this slightly? The flex lock-up used on the landing page has a full `2rem` gap between buttons. I tried that here and it’s too much given the context with less vertical whitespace, but I think `1em` does work OK

| 0.5em | 1em |
|---|---|
| <img width="566" alt="image" src="https://github.com/user-attachments/assets/b1323b1d-8d02-4956-ae0f-73ee5d571ca3"> | <img width="574" alt="image" src="https://github.com/user-attachments/assets/32a9321d-5316-47ad-b533-ee841cfea5ef"> |

It’s particularly helpful for the minimal variant, which felt like it was a bit squished up too close to the other links otherwise.

```suggestion
		margin-inline-end: 1em;
```

Reply:
I quite like it too on desktop. On mobile, I'm less confident but I pushed the change and tweak if needed on mobile depending on what you think.